templates/experiments/chat/input_bar.html

```diff
@@ -20,55 +20,59 @@
   <div class="flex flex-col" {% if assistant %}x-data="fileUploads"{% endif %}>
     <div class="flex items-center w-full">
       {% if assistant %}
-        <div class="mt-1 flex flex-row gap-2">
-          <div class="dropdown dropdown-top">
-            <div tabindex="0" role="button" class="btn m-1"><i class="fa-solid fa-paperclip"></i></div>
-            <ul tabindex="0" class="dropdown-content menu bg-base-100 rounded-box z-[1] w-52 p-2 shadow">
-              <li>
-                <a
-                  class="btn btn-sm"
-                  {% if not assistant.supports_code_interpreter %}
-                    disabled
-                  {% else %}
-                    x-bind:disabled="disableCodeInterpreterUpload"
-                  {% endif %}
-                  @click="$refs.codeInterpreter.click()">
-                  Code Interpreter
-                </a>
-              </li>
-              <li>
-                <a
-                  class="btn btn-sm"
-                  {% if not assistant.supports_file_search %} disabled {% endif %}
-                  @click="$refs.fileSearch.click()">
-                  File Search
-                </a>
-              </li>
-            </ul>
-          </div>
+        {% if assistant.supports_code_interpreter and assistant.supports_file_search %}
+          <div class="flex flex-row gap-2">
+            <div class="dropdown dropdown-top">
+              <div tabindex="0" role="button" class="btn m-1"><i class="fa-solid fa-paperclip"></i></div>
+              <ul tabindex="0"
+                  class="dropdown-content menu bg-base-100 rounded-box z-[1] w-40 p-2 shadow content-start">
+                <li>
+                  <a x-bind:disabled="disableCodeInterpreterUpload" @click="$refs.codeInterpreter.click()">Code
+                    Interpreter</a>
+                </li>
+                <li>
+                  <a @click="$refs.fileSearch.click()">File Search</a>
+                </li>
+              </ul>
+            </div>
+            {% partialdef code-interpreter-input inline=True %}
```

Comment:
I'm using 'inline' partials so that they render here and can also be referenced elsewhere

Reply:
Nice so if without that flag the definition doesn't render it until called later on? Neat